sdcm/teardown_validators/protected_db_nodes_memory.py

```diff
@@ -0,0 +1,201 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright (c) 2025 ScyllaDB
+
+import re
+import time
+import logging
+from pathlib import Path
+from datetime import datetime, timezone
+from typing import List, Tuple, Union
+
+from argus.client.sct.types import LogLink
+
+from sdcm.cluster import BaseNode
+from sdcm.utils.common import S3Storage
+from sdcm.logcollector import GrafanaScreenShot
+from sdcm.monitorstack.ui import DetailedLsaTotalMemory
+from sdcm.sct_events import Severity
+from sdcm.sct_events.teardown_validators import ValidatorEvent
+from sdcm.teardown_validators.base import TeardownValidator
+from sdcm.utils.validators.anomalies_detection import detect_isolation_forest_anomalies, bytes_to_readable
+from sdcm.utils.decorators import silence
+from sdcm.sct_events.file_logger import get_events_grouped_by_category
+
+
+LOG = logging.getLogger(__name__)
+
+
+class ProtectedDbNodesMemoryValidator(TeardownValidator):
```

Comment:
I think we should describe here the purpose and how we do it (mentioning what are protected db nodes and what anomalies we are targeting and why)